homeassistant/components/esphome/light.py

```diff
@@ -281,7 +349,7 @@ def supported_features(self) -> int:
     @property
     def supported_color_modes(self) -> set[str] | None:
         """Flag supported color modes."""
-        return set(map(_COLOR_MODES.from_esphome, self._native_supported_color_modes))
+        return set(map(_color_mode_to_ha, self._native_supported_color_modes))
```

Comment:
There are some rules for the color modes that must be followed:
```
COLOR_MODE_ONOFF = "onoff"  # Must be the only supported mode
COLOR_MODE_BRIGHTNESS = "brightness"  # Must be the only supported mode
COLOR_MODE_WHITE = "white"  # Must *NOT* be the only supported mode
```
Is that guaranteed to hold true? If not the set should be filtered before we return it.
There's a helper `valid_supported_color_modes` which does all the necessary tests, although it throws instead of filters.